presto-hive/src/main/java/com/facebook/presto/hive/HiveType.java

```diff
@@ -271,4 +284,38 @@ public static Type getPrimitiveType(PrimitiveTypeInfo typeInfo)
                 return null;
         }
     }
+
+    public static void checkForceBigintWritableType(TypeInfo typeInfo)
+    {
+        switch (typeInfo.getCategory()) {
+            case PRIMITIVE:
+                PrimitiveObjectInspector.PrimitiveCategory primitiveCategory = ((PrimitiveTypeInfo) typeInfo).getPrimitiveCategory();
+                checkForceBigintWritableType(primitiveCategory);
+                break;
+            case MAP:
+                MapTypeInfo mapTypeInfo = checkType(typeInfo, MapTypeInfo.class, "typeInfo");
+                checkForceBigintWritableType(mapTypeInfo.getMapKeyTypeInfo());
+                checkForceBigintWritableType(mapTypeInfo.getMapValueTypeInfo());
+                break;
+            case LIST:
+                ListTypeInfo listTypeInfo = checkType(typeInfo, ListTypeInfo.class, "typeInfo");
+                checkForceBigintWritableType(listTypeInfo.getListElementTypeInfo());
+                break;
+            case STRUCT:
+                StructTypeInfo structTypeInfo = checkType(typeInfo, StructTypeInfo.class, "typeInfo");
+                structTypeInfo.getAllStructFieldTypeInfos().forEach(HiveType::checkForceBigintWritableType);
+                break;
+        }
+    }
+
+    private static void checkForceBigintWritableType(PrimitiveObjectInspector.PrimitiveCategory primitiveCategory)
+    {
+        switch (primitiveCategory) {
+            case INT:
+            case SHORT:
+            case BYTE:
+                throw new IllegalArgumentException("disabled type: " + primitiveCategory);
```

Comment:
This should be a `PrestoException` probably with the `NOT_SUPPORTED` error code. Unless this indicates a bug in the code, which it doesn't look like